packages/types/lib/utils.ts

```diff
@@ -12,3 +12,9 @@ export interface Logger {
 
 type ValidateSelection<T, U> = U extends T ? U : never;
 export type PickFromUnion<T, U extends T> = ValidateSelection<T, U>;
+
+export type NullablePartial<
+    TBase,
+    NK extends keyof TBase = { [K in keyof TBase]: null extends TBase[K] ? K : never }[keyof TBase],
+    NP = Partial<Pick<TBase, NK>> & Pick<TBase, Exclude<keyof TBase, NK>>
+> = { [K in keyof NP]: NP[K] };
```

Comment:
not sure you need NP. can we have `... = Omit<TBase, TNullableKeys> & Partial<Pick<TBase, TNullableKeys>>`

Reply:
You forgot to add the non nullable keys, but yes I can simplify indeed